scheduler/log.c

```diff
@@ -165,9 +165,10 @@ cupsdCheckLogFile(cups_file_t **lf,	/* IO - Log file */
 	{
 	 /*
 	  * Insert the server name...
+	  * If it is NULL use "localhost" to avoid core dump.
 	  */
 
-	  cupsCopyString(ptr, ServerName, sizeof(filename) - (size_t)(ptr - filename));
+	  cupsCopyString(ptr, ServerName == 0 ? "localhost" : ServerName, sizeof(filename) - (size_t)(ptr - filename));
```

Comment:
Shouldn't be comparing to 0, either just use "ServerName" with the implicit NULL == false or compare to NULL. Also, in other places we generally have the more likely case first, e.g.:

```c
cupsCopyString(ptr, ServerName ? ServerName : "localhost", sizeof(filename) - (size_t)(ptr - filename));
```